### PR TITLE
Fixed #27882 -- Allowed {% cache %} to cache indefinitely.

### PR DESCRIPTION
**Pull request by rlmv**
Fixed the cache templatetag so that fragments will be cached
forever when the cache timeout is set to None.

### CODE REVIEW

File: django/templatetags/cache.py [hunk 1]
```diff
@@ -20,10 +20,13 @@ def render(self, context):
             expire_time = self.expire_time_var.resolve(context)
         except VariableDoesNotExist:
             raise TemplateSyntaxError('"cache" tag got an unknown variable: %r' % self.expire_time_var.var)
-        try:
-            expire_time = int(expire_time)
-        except (ValueError, TypeError):
-            raise TemplateSyntaxError('"cache" tag got a non-integer timeout value: %r' % expire_time)
+
```
Comment (timgraham) [inline django/templatetags/cache.py]: The added blank lines aren't needed.

File: docs/topics/cache.txt [hunk 2]
```diff
@@ -715,6 +715,8 @@ equivalent:
 This feature is useful in avoiding repetition in templates. You can set the
 timeout in a variable, in one place, and just reuse that value.
 
+If you set the cache timeout to ``None`` the fragment will never expire.
```
Comment (timgraham) [inline docs/topics/cache.txt]: Add a `.. versionchanged:: 2.0` annotation.

File: tests/template_tests/syntax_tests/test_cache.py [hunk 3]
```diff
@@ -122,6 +122,14 @@ def test_cache18(self):
         output = self.engine.render_to_string('cache18')
         self.assertEqual(output, 'cache18')
 
+    @setup({'cache19': '{% load cache %}{% cache None test %}cache19{% endcache %}'})
+    def test_cache19(self):
+        """
+        Allow cache timeout to be None (#27882).
```
Comment (timgraham) [inline tests/template_tests/syntax_tests/test_cache.py]: Ticket reference isn't needed. The docstring could be omitted in place of a meaningful test name, e.g. `test_timeout_none`. Could set test be added to check that `cache.set()` is called with `None`, i.e. that the value is actually cached?

Comment (rlmv) [inline tests/template_tests/syntax_tests/test_cache.py]: Yep, I can check that the content makes it into the cache. FYI it looks like most of the other cache tag tests don't test this either; maybe that can be fixed in another PR.

**Comment by rlmv** [comment]
Requested changes made and rebased on master.

File: docs/topics/cache.txt [hunk 4]
```diff
@@ -715,6 +715,10 @@ equivalent:
 This feature is useful in avoiding repetition in templates. You can set the
 timeout in a variable, in one place, and just reuse that value.
 
+.. versionchanged:: 2.0
+
+    If the cache timeout is set to ``None`` the fragment will never expire.
```
Comment (srinivasreddy) [inline docs/topics/cache.txt]: or will forever be cached?

Comment (rlmv) [inline docs/topics/cache.txt]: Yeah, I suppose that would be a little more clear.

**Pull request merged**